packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/BaseModelSqlv2.ts

```diff
@@ -2249,17 +2251,19 @@ class BaseModelSqlv2 {
               ? customValidators[func[j]]
               : Validator[func[j]]
             : func[j];
+        const columnValue = columns?.[cn] || columns?.[columnTitle];
         const arg =
-          typeof func[j] === 'string' ? columns[cn] + '' : columns[cn];
+          typeof func[j] === 'string' ? columnValue + '' : columnValue;
         if (
-          columns[cn] !== null &&
-          columns[cn] !== undefined &&
-          columns[cn] !== '' &&
-          cn in columns &&
-          !(fn.constructor.name === 'AsyncFunction' ? await fn(arg) : fn(arg))
+          !(fn.constructor.name === 'AsyncFunction'
+            ? await fn(arg)
+            : fn(arg)) &&
+          ![null, undefined, ''].includes(columnValue)
```

Comment:
I think here first we need to check value is null or undefined and then we need to invoke the validator method

```js
 ![null, undefined, ''].includes(columnValue) 
      && !(fn.constructor.name === 'AsyncFunction' ? await fn(arg) : fn(arg))
```

Reply:
Done